model/task_lifecycle.go

```diff
@@ -906,21 +916,27 @@ func updateBuildGithubStatus(b *build.Build, buildTasks []task.Task) error {
 	return b.UpdateGithubCheckStatus(githubBuildStatus)
 }
 
-// UpdateBuildStatus updates the status of the build based on its tasks' statuses.
+// updateBuildStatus updates the status of the build based on its tasks' statuses.
 // Returns true if the build's status has changed.
-func UpdateBuildStatus(b *build.Build) (bool, error) {
+func updateBuildStatus(b *build.Build) (bool, error) {
 	buildTasks, err := task.FindWithFields(task.ByBuildId(b.Id), task.StatusKey, task.ActivatedKey, task.DependsOnKey, task.IsGithubCheckKey, task.AbortedKey)
 	if err != nil {
 		return false, errors.Wrapf(err, "getting tasks in build '%s'", b.Id)
 	}
 
-	buildStatus := getBuildStatus(buildTasks)
+	buildStatus, allTasksBlocked := getBuildStatus(buildTasks)
+
+	if allTasksBlocked != b.AllTasksBlocked {
+		if err = b.SetBlocked(allTasksBlocked); err != nil {
+			return false, errors.Wrapf(err, "setting build '%s' as blocked", b.Id)
+		}
+	}
 
 	if buildStatus == b.Status {
-		return false, nil
+		return allTasksBlocked != b.AllTasksBlocked, nil
```

Comment:
ah youre right, i still added a boolean to check if there has been a change because we would still want to update the version status if the build status goes from "created, not blocked" to "created, blocked"